lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -44,12 +49,27 @@ def self.fabricpath_feature
       fail 'fabricpath_feature not found' if fabricpath.nil?
       return :disabled if fabricpath.nil?
       fabricpath.to_sym
+    rescue Cisco::CliError => e
+      # cmd will syntax reject when feature is not enabled
+      raise unless e.clierror =~ /Syntax error/
+      return :disabled
     end
 
     def self.fabricpath_feature_set(fabricpath_set)
       curr = FabricpathGlobal.fabricpath_feature
       return if curr == fabricpath_set
 
+      if Vdc.vdc_support
+        # For modular platforms, make sure to limit the feature to
+        # modules which support this feature
+        if fabricpath_set == :installed || fabricpath_set == :enabled
+          v = Vdc.new('default')
+          v.limit_resource_module_type =
+            config_get('fabricpath', 'supported_modules')
```

Comment:
This feels very dangerous to me. 

If a user has non-F3 (or other non-fabric) modules and they set their manifest to use fabricpath then all of the interfaces are going to disappear from the vdc. I would think that this would be very unexpected from a user perspective. 

I think a better approach is to have the user specify the modules, which they can already do with cisco_vdc. At least then they know what they're getting into.

This raises another question though: After the limit-resource command is used, don't you also have to allocate interfaces to the vdc?

I would be open to having fabricpath trigger both of these actions but I would still want some user control over them; ie.

```
cisco_fabricpath_global {
  limit_resource_module_type => 'f3',
  allocate_interface => 'all',
}
```